### PR TITLE
Add console warning when legacy colors are used

### PR DESCRIPTION
**Pull request by darkwing**
Not sure how immediately useful this is, if at all, for identifying colors we don't want to be using anymore. Might be useful during dark mode-recoloring.

### CODE REVIEW

File: ui/components/ui/box/README.mdx [hunk 1]
```diff
@@ -107,18 +107,10 @@ GREY: 'grey',
...
-SECONDARY3: 'secondary-3',
 SUCCESS1: 'success-1',
-SUCCESS2: 'success-2',
 SUCCESS3: 'success-3',
 ERROR1: 'error-1',
-ERROR2: 'error-2',
-ERROR3: 'error-3',
 ALERT1: 'alert-1',
-ALERT2: 'alert-2',
-ALERT3: 'alert-3',
```
Comment (georgewrmarshall) [inline ui/components/ui/box/README.mdx:123]: Do these remove colors that aren't used at all?

**Review by georgewrmarshall — APPROVED** [review]
This is awesome! It would be cool if our tests failed for any new code that used incorrect prop types...

File: ui/components/ui/box/README.mdx (hunk 1 above)
Comment (darkwing) [reply ui/components/ui/box/README.mdx:123]: Yep; I've done searches and some of these aren't even used.

**Comment by darkwing** [comment]
Closing because https://github.com/MetaMask/metamask-extension/pull/13885 mostly nullifies this.

**Pull request closed without merging**